windows-agent/internal/proservices/landscape/executor.go

```diff
@@ -189,3 +196,119 @@ func (e executor) setDefault(ctx context.Context, cmd *landscapeapi.Command_SetD
 func (e executor) shutdownHost(ctx context.Context, cmd *landscapeapi.Command_ShutdownHost) error {
 	return gowsl.Shutdown(ctx)
 }
+
+func installFromMicrosoftStore(ctx context.Context, distro gowsl.Distro) error {
+	if err := gowsl.Install(ctx, distro.Name()); err != nil {
+		return err
+	}
+
+	if err := distroinstall.InstallFromExecutable(ctx, distro); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func installFromURL(ctx context.Context, homeDir string, distro gowsl.Distro, rootfs *landscapeapi.Command_Install_Rootfs) (err error) {
+	// Since we are going to remove the tarball soon, I'd say it's fine to write it to a subfolder of the current working directory,
+	// as it will be somewhere inside the Windows Apps private folders:
+	// %USERPROFILE%\AppData\Local\Packages\CanonicalGroupLimited.Ubuntu_79rhkp1fndgsc\LocalState\
```

Comment:
```suggestion
	// %USERPROFILE%\AppData\Local\Packages\CanonicalGroupLimited.UbuntuPro_79rhkp1fndgsc\LocalState\
```

When I wrote that comment I didn't mean it to stay, but I guess it's useful, so let's just correct this path.